### ISSUE TITLE
issue about json compress

### ISSUE THREAD

**Issue by truongca1902** [minio/minio#18764]
Hi I have a problem about rolling back the version of file  .json as follows
I uploaded a file with a capacity of 47kib
Then I restored it and found it was only 6.9kib and when I got the file it couldn't be read
This problem was only resolved when I turned off the json file compression feature (but only applies to uploaded files after turning off the feature).
Is there any other solution or is this a bug of minio
![image](https://github.com/minio/minio/assets/78651581/5c414808-6bfc-476f-8c62-e4096daf0fba)

**Comment by klauspost** [comment]
What do you mean by "when I turned off the json file compression feature".

Also please list your complete setup, server versions as requested by the template.

**Comment by truongca1902** [comment]
> server versions as requested by the template.
this is my setup 
```
MINIO_VOLUMES="http://minio{1...4}.com:9000/mnt/data{1...4}/ http://minio{5...8}.com:9000/mnt/data{1...4}/"
MINIO_OPTS="--console-address :9001"

MINIO_ROOT_USER=******


MINIO_ROOT_PASSWORD=*****************

MINIO_SERVER_URL="https://minio.local.vn"
```
 my version
![image](https://github.com/minio/minio/assets/78651581/5f42c650-25cf-4b14-a9e4-7a0574a1a2c3)
my config turn off compress json
![image](https://github.com/minio/minio/assets/78651581/1eab1232-96cb-4fe5-9afe-d89643c443c5)
data file content after restore
![image](https://github.com/minio/minio/assets/78651581/47814a45-96eb-49c2-ba11-d3ddd18d62c8)

**Comment by klauspost** [comment]
ok, so when you say you "restored it" - what did you do exactly?

**Comment by truongca1902** [comment]
> ok, so when you say you "restored it" - what did you do exactly?
I performed the restore on the GUI 
![image](https://github.com/minio/minio/assets/78651581/bd18b8ad-8e2f-41bd-9449-5742aeb2748f)
and 
and you can see that the two versions have different sizes
![image](https://github.com/minio/minio/assets/78651581/c0241b2a-9e84-45be-9c35-7ea9e25302f3)

**Comment by klauspost** [comment]
OK. Thanks! I will see if I can replicate and see what is happening.

**Comment by jiuker** [comment]
fixed by #19895
